compiler/ballerina-lang/src/main/java/io/ballerina/projects/PackageResolution.java

```diff
@@ -485,14 +486,56 @@ private BlendedManifest createBlendedManifest(PackageContext rootPackageContext,
 
     private ResolutionOptions getResolutionOptions(PackageContext rootPackageContext,
                                                    CompilationOptions compilationOptions) {
+        boolean sticky = getSticky(rootPackageContext);
+        PackageLockingMode packageLockingMode;
+        SemanticVersion prevDistributionVersion = rootPackageContext.dependencyManifest().distributionVersion();
+        SemanticVersion currentDistributionVersion = SemanticVersion.from(RepoUtils.getBallerinaShortVersion());
+        if (!sticky && (null == prevDistributionVersion
+                || isNewUpdateDistribution(prevDistributionVersion, currentDistributionVersion))) {
+            packageLockingMode = PackageLockingMode.SOFT;
+            if (rootPackageContext.dependenciesTomlContext().isPresent()) {
+                addOlderSLUpdateDistributionDiagnostic(prevDistributionVersion, currentDistributionVersion);
+            }
+        } else {
+            packageLockingMode = PackageLockingMode.MEDIUM;
+        }
         return ResolutionOptions.builder()
                 .setOffline(compilationOptions.offlineBuild())
-                .setSticky(getSticky(rootPackageContext))
+                .setSticky(sticky)
                 .setDumpGraph(compilationOptions.dumpGraph())
                 .setDumpRawGraphs(compilationOptions.dumpRawGraphs())
+                .setPackageLockingMode(packageLockingMode)
                 .build();
     }
 
+    private boolean isNewUpdateDistribution(SemanticVersion prevDistributionVersion,
+                                                  SemanticVersion currentDistributionVersion) {
+        return currentDistributionVersion.major() == prevDistributionVersion.major()
+                && currentDistributionVersion.minor() > prevDistributionVersion.minor();
+    }
+
+    private void addOlderSLUpdateDistributionDiagnostic(SemanticVersion prevDistributionVersion,
+                                                        SemanticVersion currentDistributionVersion) {
+        String prevVersionForDiagnostic;
+        String currentVersionForDiagnostic =
+                currentDistributionVersion.minor() + DOT + currentDistributionVersion.patch();
+        if (null != prevDistributionVersion) {
+            prevVersionForDiagnostic = prevDistributionVersion.minor()  + DOT + prevDistributionVersion.patch();
+        } else {
+            prevVersionForDiagnostic = "4 or an older Update";
+        }
+        DiagnosticInfo diagnosticInfo = new DiagnosticInfo(
+                ProjectDiagnosticErrorCode.BUILT_WITH_OLDER_SL_UPDATE_DISTRIBUTION.diagnosticId(),
+                "Detected an attempt to build this package using Swn Lake Update " + currentVersionForDiagnostic +
```

Comment:
```suggestion
                "Detected an attempt to build this package using Swan Lake Update " + currentVersionForDiagnostic +
```

Reply:
Let's  append the patch version for only the patch distributions. `Swan Lake Update 5.0` looks a bit odd.